.azure/pipelines/ci.yml

```diff
@@ -780,12 +762,62 @@ stages:
             - Linux_musl_x64_build
             - Linux_musl_arm_build
             - Linux_musl_arm64_build
-          # In addition to the dependencies above, ensure the build was successful overall.
+          # In addition to the dependencies above that provide assets, ensure the build was successful overall.
+          - ${{ if in(variables['Build.Reason'], 'Manual') }}:
+            - Code_check
+            - Windows_Test
+            - MacOS_Test
+            - Linux_Test
+            - Helix_x64
           - Source_Build_Managed
         pool:
           vmImage: vs2017-win2016
         publishUsingPipelines: ${{ variables._PublishUsingPipelines }}
         enablePublishBuildArtifacts: true # publish artifacts/log files
+  - ${{ and(ne(variables['System.TeamProject'], 'public'), eq(variables['Build.SourceBranch'],'refs/heads/main')) }}:
+    - template: /eng/common/templates/job/source-index-stage1.yml
+      parameters:
+        binlogPath: $(System.ArtifactsDirectory)/Linux_x64_Logs/Release/Build.binlog
+        dependsOn:
+          - Linux_x64_build
+          # In addition to the dependency above that provides assets, ensure the build was successful overall.
+          - Windows_build
+          - Windows_arm_build
+          - Windows_arm64_build
+          - ${{ if ne(variables['PostBuildSign'], 'true') }}:
+            - CodeSign_Xplat_MacOS_arm64
+            - CodeSign_Xplat_MacOS_x64
+            - CodeSign_Xplat_Linux_x64
+            - CodeSign_Xplat_Linux_arm
+            - CodeSign_Xplat_Linux_arm64
+            - CodeSign_Xplat_Linux_musl_x64
+            - CodeSign_Xplat_Linux_musl_arm
+            - CodeSign_Xplat_Linux_musl_arm64
+          - ${{ if eq(variables['PostBuildSign'], 'true') }}:
+            - MacOs_arm64_build
+            - MacOs_x64_build
+            - Linux_arm_build
+            - Linux_arm64_build
+            - Linux_musl_x64_build
+            - Linux_musl_arm_build
+            - Linux_musl_arm64_build
+          - ${{ if in(variables['Build.Reason'], 'Manual') }}:
+            - Code_check
+            - Windows_Test
+            - MacOS_Test
+            - Linux_Test
+            - Helix_x64
```

Comment:
This list is a little spooky, it seems likely people won't know to update it if/when our set of jobs changes

Reply:
Been this way forever unfortunately. AzDO schema doesn't provide a way to automagically wait for almost every other job.

Main mitigation is @dotnet/aspnet-build are automagically reviewers whenever a .azure/ file is changed.